### PR TITLE
ENH: show_bbox and drop_id added to geoseries.to_json

### LINKED ISSUE #3199: ENH: parity between GeoDataFrame.to_json and GeoSeries.to_json
> Hey, guys @jdmcbr @martinfleis , just bringing to your attention that while you fixed this issue for GeoDataFrame, id field is still obligatory for GeoSeries, there is no option to remove it. You might want to add it for GeoSeries as well.
>
> My version of geopandas is 0.14.2.

_Originally posted by @kiranzo in https://github.com/geopandas/geopandas/issues/1637#issuecomment-1958551545_
            

Some attributes shall be included in `GeoSeries.to_json` as well.

### PR DESCRIPTION
**Pull request by nicholas-ys-tan**
Closes #3199 using existing functions in GeoDataFrame.

Have not included `na` as an attribute as Properties aren't used in GeoSeries as far as I understand.

I noted default of `GeoDataFrame.to_json` show_bbox is False, but the existing code defaults to True, hence have explicitly left default as True.

### CODE REVIEW

**Comment by nicholas-ys-tan** [comment]
Hello, have been looking to contribute to geopandas as a user of this repository - this seemed like a good low hanging fruit to get my feet wet with my first PR.

I've read through the contribution guide and hope I have not violated any of the conventions. Kindly let me know if I did.

Thanks.

File: geopandas/geoseries.py [hunk 1]
```diff
@@ -1189,12 +1189,18 @@ def estimate_utm_crs(self, datum_name: str = "WGS 84"):
         """
         return self.values.estimate_utm_crs(datum_name)
 
-    def to_json(self, **kwargs) -> str:
+    def to_json(self, show_bbox=True, drop_id=False, **kwargs) -> str:
```
Comment (m-richards) [inline geopandas/geoseries.py:1192]: was their any reason in your mind why not to also expose the `to_wgs84`? It seems if we are adding this we may as well cover all keywords.

File: geopandas/geoseries.py [hunk 2]
```diff
@@ -1223,7 +1229,13 @@ def to_json(self, **kwargs) -> str:
         --------
         GeoSeries.to_file : write GeoSeries to file
         """
-        return json.dumps(self.__geo_interface__, **kwargs)
+        from geopandas import GeoDataFrame
+
+        return json.dumps(
```
Comment (m-richards) [inline geopandas/geoseries.py:1234]: we should also still pass **kwargs through to json.dumps.

**Review by m-richards — COMMENTED** [review]
Thanks @nicholas-ys-tan for making a start on this!

I left two small comments, but I have a general suggestion, I think you could implement GeoSeries.to_json directly in terms of GeoDataFrame.to_json, which would expose `to_wgs84` eve - perhaps you could investigate whether `to_json` should be defined on the parent class of GeoSeries/GeoDataFrame, `geopandas/base.py::GeoPandasBase`. We do this for a lot of other methods for code sharing, and it looks to me like exposing all the options from GeoDataFrame.to_json into GeoSeries.to_json will end up replicating all the code.

As you note, the default values for `show_bbox` are currently inconsistent for GeoSeries vs GeoDataFrame, so we would still need an overload for GeoSeries. 
You might find it easier to first implement the GeoSeries method in terms of the GeoDataFrame one and checking this works before moving anything to the parent class.

File: geopandas/geoseries.py (hunk 2 above)
Comment (nicholas-ys-tan) [reply geopandas/geoseries.py:1234]: Thanks Matt, I missed that

File: geopandas/geoseries.py (hunk 1 above)
Comment (nicholas-ys-tan) [reply geopandas/geoseries.py:1192]: No good reason, should have been included upon review, thanks

**Comment by nicholas-ys-tan** [comment]
Thank you for taking the time to review and provide detailed feedback @m-richards 

Agreed with both of your comments and have updated accordingly. Using the `to_json` method directly from geodataframe certainly addresses both of your comments without replicating the code for crs transformations.

Regarding your more general comment on structure, I had a look at putting the function in `geopandas/base.py::GeoPandasBase` but the functions currently in the base class seem to be more on properties and geometric functions. The other GeoSeries i/o functions `to_file`, `from_file` inherit from `geodataframe` so I was seeking to mimic that structure in the latest commit. I am happy to move it to the base class if you think that is where it belongs though.

That said, I was thinking about moving the `to_json` method into a new file in `geopandas/io/json.py` as `_to_json` to facilitate working on this issue: https://github.com/geopandas/geopandas/issues/3187 in the future, it would also mimic the structure of `to_file`, but thought maybe that's best done as part of issue 3187, if I look at working on that down the road being a more complex one.

Again, appreciate your time and feedback.

Noted there's a failing test - I'll keep looking into it

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,11 @@ New features and improvements:
   the specified columns (#3101).
 - Added support to ``read_file`` for the ``mask`` keyword for the pyogrio engine (#3062).
 - Added support to ``read_file`` for the ``columns`` keyword for the fiona engine (#3133).
+- Added `show_bbox` and `drop_id` arguments to allow removal of unwanted parameters from
```
Comment (martinfleis) [inline CHANGELOG.md:51]:
```suggestion
- Added `show_bbox`, `drop_id` and `to_wgs84` arguments to allow further customization of 
```

File: geopandas/geoseries.py [hunk 4]
```diff
@@ -1190,12 +1189,18 @@ def estimate_utm_crs(self, datum_name: str = "WGS 84"):
         """
         return self.values.estimate_utm_crs(datum_name)
 
-    def to_json(self, **kwargs) -> str:
+    def to_json(self, show_bbox=True, drop_id=False, to_wgs84=False, **kwargs) -> str:
         """
         Returns a GeoJSON string representation of the GeoSeries.
 
         Parameters
```
Comment (martinfleis) [inline geopandas/geoseries.py:1196]: `to_wgs84` is missing from the docstring.

**Review by martinfleis — COMMENTED** [review]
Two minor documentation-related notes.

File: geopandas/geoseries.py (hunk 4 above)
Comment (nicholas-ys-tan) [reply geopandas/geoseries.py:1196]: apologies, have fixed that

**Review by martinfleis — APPROVED** [review]
Thanks!

File: geopandas/geoseries.py [hunk 5]
```diff
@@ -1190,12 +1189,25 @@ def estimate_utm_crs(self, datum_name: str = "WGS 84"):
         """
         return self.values.estimate_utm_crs(datum_name)
 
-    def to_json(self, **kwargs) -> str:
+    def to_json(self, show_bbox=True, drop_id=False, to_wgs84=False, **kwargs) -> str:
```
Comment (m-richards) [inline geopandas/geoseries.py:1192]:
```suggestion
    def to_json(self, show_bbox:bool=True, drop_id:bool=False, to_wgs84:bool=False, **kwargs) -> str:
```
(geoseries.py is the only typing PR that has been merge so far, so it makes sense to keep types in the annotations where it's easy).

**Review by m-richards — APPROVED** [review]
Thanks @nicholas-ys-tan for this! One final small suggestion, but otherwise looks great

File: geopandas/geoseries.py (hunk 5 above)
Comment (nicholas-ys-tan) [reply geopandas/geoseries.py:1192]: Thanks Matt! I'll keep that in mind

**Pull request merged**